exchange/exchange.go

```diff
@@ -65,6 +66,12 @@ func New(ctx context.Context, h host.Host, ds datastore.Batching, opts Options)
 		rou:  NewGossipRouting(h, opts.PubSub, opts.GossipTracer, opts.Regions),
 	}
 
+	// remove unwanted blocks that might be in the blockstore but are removed from the index
+	err = exch.Index().CleanBlockStore()
```

Comment:
Might move this later to add more flexibility when implementing the Exchange but for now it should work fine.

Reply:
I will put it in popn & popp